llama-index-core/llama_index/core/workflow/context.py

```diff
@@ -49,6 +49,7 @@ def __init__(
         )
         self._accepted_events: List[Tuple[str, str]] = []
         self._retval: Any = None
+        self._in_progress: Dict[str, List[Event]] = defaultdict(list)
```

Comment:
We need to keep track of steps that are in progress at the creation time of a checkpoint. Otherwise, since their input events have already been popped from their respective queues, we will lose these events if we attempt to load from a checkpoint.